File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/TextDisplayActivity.java
package at.tugraz.tc.cyfile.ui;

import android.os.Bundle;
import android.widget.TextView;

import at.tugraz.tc.cyfile.R;

public class TextDisplayActivity extends BaseActivity {

    public static final String ABOUT = "ABOUT";
    public static final String HELP = "HELP";
    public static final String LICENSES = "LICENSES";
    public static final String CONTACT = "CONTACT";

    private static String aboutString;
    private static String helpString;
    private static String licensesString;
    private static String contactString;

    private TextView content;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_text_display);
        aboutString = getString(R.string.about_string);
        helpString = getString(R.string.help_string);
        licensesString = getString(R.string.licenses_string);
        contactString = getString(R.string.contact_string);
        content = findViewById(R.id.text_content);
        setText();
    }

    private void setText() {
        String content_id = getIntent().getStringExtra(SettingsActivity.CONTENT_ID);
        switch (content_id) {
            case ABOUT:
                content.setText(aboutString);
                break;
            case HELP:
                content.setText(helpString);
                break;
            case LICENSES:
                content.setText(licensesString);
                break;
            case CONTACT:
                content.setText(contactString);
                break;
            default:
                content.setText("");
                break;
        }
    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/TextDisplayActivity.kt
package at.tugraz.tc.cyfile.ui

import android.content.Intent
import android.os.Bundle
import android.widget.TextView

import at.tugraz.tc.cyfile.R

class TextDisplayActivity : BaseActivity() {

    private var content: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_text_display)
        aboutString = getString(R.string.about_string)
        helpString = getString(R.string.help_string)
        licensesString = getString(R.string.licenses_string)
        contactString = getString(R.string.contact_string)
        content = findViewById(R.id.text_content)
        setText()
    }

    private fun setText() {
        val contentId = intent.getStringExtra(SettingsActivity.CONTENT_ID)
        when (contentId) {
            ABOUT -> content!!.text = aboutString
            HELP -> content!!.text = helpString
            LICENSES -> content!!.text = licensesString
            CONTACT -> content!!.text = contactString
            else -> content!!.text = ""
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val i = Intent(this@TextDisplayActivity, SettingsActivity::class.java)
        i.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
        startActivity(i)
        finish()
    }

    companion object {

        const val ABOUT = "ABOUT"
        const val HELP = "HELP"
        const val LICENSES = "LICENSES"
        const val CONTACT = "CONTACT"

        private var aboutString: String? = null
        private var helpString: String? = null
        private var licensesString: String? = null
        private var contactString: String? = null
    }

}
